controlplane/eks/controllers/awsmanagedcontrolplane_controller.go

```diff
@@ -283,9 +283,11 @@ func (r *AWSManagedControlPlaneReconciler) reconcileDelete(ctx context.Context,
 		return reconcile.Result{}, err
 	}
 
-	if err := sgService.DeleteSecurityGroups(); err != nil {
-		log.Error(err, "error deleting general security groups for AWSManagedControlPlane", "namespace", controlPlane.Namespace, "name", controlPlane.Name)
-		return reconcile.Result{}, err
+	if conditions.GetReason(managedScope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition) != clusterv1.DeletedReason {
```

Comment:
I don't think using the condition as part of reconciliation logic is the right way to go....when in the same reconciliation process.

Would something like this work:
* change `DeleteSecurityGroup` :
  * If the length of `clustergroups` is 0 then exit early 
  * Then if length of `clustergroups` is greater than set `clusterv1.DeletingReason` 
  * In any error handling blocks just return the error
* change AWSMCP reconcileDelete:
  *  In the error handler block for `sgService.DeleteSecurityGroups()` set the `DeletingFailed` reason
  * if successful set the `DeletedReason`

Reply:
Thanks for the feedback! That sounds good. Updated logic according to your suggestion.